api/src/utils/csv-utils/row-validators/start-end-date-order-row-validator.ts

```diff
@@ -0,0 +1,64 @@
+import dayjs from 'dayjs';
+import { isDateTimeString } from '../../date-time-utils';
+import { CSVConfigUtils } from '../csv-config-utils';
+import { CSVRowValidator } from '../csv-config-validation.interface';
+
+/**
+ * Get a row validator that checks if the start date is before the end date.
+ *
+ * Rules:
+ *  1. Start date and time must be before end date and time.
+ *
+ * @template StaticHeaderType
+ * @param {CSVConfigUtils} utils - CSV config utils
+ * @param {Record<string, string>} staticHeaders - Static headers
+ * @returns {CSVRowValidator} - Row validator
+ */
+export const getStartDateIsBeforeEndDateRowValidator = <StaticHeaderType extends Uppercase<string> = Uppercase<string>>(
+  utils: CSVConfigUtils<StaticHeaderType>,
+  staticHeaders: {
+    startDate: StaticHeaderType;
+    startTime: StaticHeaderType | undefined;
+    endDate: StaticHeaderType;
+    endTime: StaticHeaderType | undefined;
+  }
+): CSVRowValidator => {
+  return (params) => {
+    let startTimeStamp = utils.getCellValue(staticHeaders.startDate, params.row) as string;
+    let endTimeStamp = utils.getCellValue(staticHeaders.endDate, params.row) as string;
+
+    if (staticHeaders.startTime) {
```

Comment:
It looks like `staticHeaders.startTime` is always truthy, and so the `startTimeStamp` is being created as `'2007-08-11 undefined'`. Same for the `endTimeStamp`

Should this `if` be checking the cell value instead?

Reply:
In this case, I was testing with a file that had no cell values for the time columns.

Reply:
Ooops brain dont work gud anymore. 

Fixed :)